x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/util/ShapeTestUtils.java

```diff
@@ -27,10 +27,13 @@
 import java.util.function.Function;
 
 import static org.elasticsearch.geo.GeometryTestUtils.linearRing;
+import static org.elasticsearch.test.ESTestCase.randomValueOtherThanMany;
 
 /** generates random cartesian shapes */
 public class ShapeTestUtils {
 
+    public static final double MIN_VALID_AREA = 1e-10;
```

Comment:
So this threshold is rather arbitrary. For testing purposes this is probably fine though, and this is a test-only class. However, if you want something better we could use something closer to the actual rounding error between float/double which I think could be sufficient.